magentoerpconnect/product.py

```diff
@@ -491,8 +491,14 @@ def categories(self, record):
                 main_categ_id = default_categ.id
 
         result = {'categ_ids': [(6, 0, category_ids)]}
-        if main_categ_id:  # OpenERP assign 'All Products' if not specified
-            result['categ_id'] = main_categ_id
+        # OpenERP assign 'All Products' if not specified
+        # skip main cat assignment if the current main category is already in
+        # categ_ids
+        if main_categ_id:
+            if self.backend_record.categ_id.id not in category_ids):
+                result['categ_id'] = main_categ_id
+            else:
```

Comment:
why would you need this? This branch causes the value to be overwritten with itself